dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/map/widgets/D3Map.java

```diff
@@ -0,0 +1,260 @@
+        D3 legendGroup = d3Selection.append("g")
+                                    .attr("transform", translate);
+        
+        legendGroup.append("text")
+        .attr("class", "map-legend-caption")
+        .attr("x", 0)
+        .attr("y", titleSize * -1)
+        .text(conf.getTitle());
+        
+        D3 legendValuesGroup = legendGroup.append("g");
+        
+        legendValuesGroup.selectAll("rect")
+                                       .data(colorScale.range())
+                                       .enter().append("rect")
+                                         .attr("height", legendSquareSize)
+                                         .attr("width", legendSquareSize)
+                                         .attr("y", (d, i, el) ->  rectPos.getAndAdd(legendSquareSize))
+                                         .attr("fill", (d, i, el) -> d)
+                                         .append("svg:title").text((d, i, el) -> {
+                                             return String.join(" - ", getFormattedBoundaryValues((String) d));
+                                         });
+        legendValuesGroup.selectAll("text")
+                                      .data(colorScale.range())
+                                      .enter().append("text")
+                                       .attr("class", "map-legend-val")
+                                       .attr("x", legendSquareSize + 2)
+                                       .attr("y", (d, i, el) ->  textPos.getAndAdd(legendSquareSize))
+                                       .text((d, i, el) -> {
+                                           String[] values = getFormattedBoundaryValues((String) d);
+                                           return "< " + values[1];
+                                        });
+    }
+    
+    private void fillRegions(D3 map) {
+        JSONArray countriesFeatures = geoData.countries();
+        map.style("fill", (d, i, el) -> {
+            Optional<Double> val = geoData.valueByCountry(data, countriesFeatures.get(i));
+            d3.select(el[i]).attr("class", "fill-region");
+            if (val.isPresent()) {
+                return colorScale.call(colorScale, val.get());
+            } 
+            return null;
+        }).on("mouseenter", (d, i, el) -> {
+            Optional<Double> val = geoData.valueByCountry(data, countriesFeatures.get(i));
+            String countryName = geoData.countryName(countriesFeatures.get(i));
+            mapTooltip.show(countryName, conf.getTitle(), val, conf.getFormatter());
+            return null;
+        }).on("mousemove", (d, i, el) -> {
+            mapTooltip.move();
+            return null;
+        }).on("mouseleave", (d, i, el) -> {
+            mapTooltip.hide();
+            return null;
+        }).on("click", (d, i, el) -> {
+            Optional<Entry<String, Double>> val = geoData.entryByCountry(data, countriesFeatures.get(i));
```

Comment:
Nitpick: instead of
```java
Optional<Something> myoptional = ...
if (myoptional.isPresent()) {
Something value = myoptional.get()
/*do something with value*/
}
```

It's often more lightweight to use 
```java
myoptional.ifPresent(value -> {/*do something with value*/});
```
when you don't need to return value